### ISSUE TITLE
Naming conventions yields a false positive while inspect static declaration  

### ISSUE THREAD

**Issue by la10736** [intellij-rust/intellij-rust#737]
```rust
    static mut last: u32 = 0;
```
Yield

> Static constant `last` should have an upper case name such as `LAST`

I compiled plugin this morning.

Thanks

**Comment by matklad** [comment]
Statics usually use SCREAMING_SNAKE_CASE:

```
warning: static constant `x` should have an upper case name such as `X`, #[warn(non_upper_case_globals)] on by default
 --> main.rs:1:1
  |
1 | static x: u32 = 0;
  | ^^^^^^^^^^^^^^^^^^

```

Do you have something like `allow(non_upper_case_globals)` as well? We don't support this yet, but we plan to.

**Comment by la10736** [comment]
No, I don't have any `allow(non_upper_case_globals)`. While I'm compiling I don't receive any warning. Anyway that isn't a *constant* but a **mutable** static variable.... I know that is strange to use a mutable static variable but that is the case.

**Comment by matklad** [comment]
Hm, interesting, so `rustc` does not enforce *any* naming convention for `static mut`... Sounds like a bug in `rustc`! Will investigate this.

A somewhat orthogonal issue is that there is a mild desire to remove `static mut` from the language altogether (pre rfc: https://internals.rust-lang.org/t/pre-rfc-remove-static-mut/1437/20), because `mut` usually means "unaliased", and statics are always aliased.

**Comment by matklad** [comment]
rustc issue: https://github.com/rust-lang/rust/issues/37145

**Comment by alygin** [comment]
@la10736, you can disable the inspection if you don't want to be warned about violation of this naming convention. Just go to Preferences &rarr; Editor &rarr; Inspections &rarr; Rust &rarr; Naming Conventions and switch off the "Static constant naming convention".

**Comment by la10736** [comment]
Thanks a lot for the feedback... I filed the issue just to help you, but I already know how to disable it.

**Comment by matklad** [comment]
Turns out this is indeed an bug in rustc: it should have produced a warning for lowercase static regardless of mutability. So the bug will be fixed there, and IntelliJ Rust is correct! ;) 

Thanks @la10736 for the report and @alygin for the implementation of the original inspection,
